File: _tech/algo/dynamic/jump2.go

func jump(nums []int) int {
    target := len(nums) - 1
    j := 0
    for (target != 0){
        for i:= 0; i<target; i++{
            if i+nums[i] >= target{
                j++ 
                target = i 
            } 
        }
    }
    
    return j 

}

File: _arch/algo/dynamic/jump2.java
  /*
  reach the last index in the minimum number of jumps.
  Input: [2,3,1,1,4]
  Output: 2
  */
  public int jump(int[] nums) {
    int steps = 0;
    int position = nums.length - 1;
        
    while(position != 0){
        for(int i=0;i<position;i++){
            if(i+nums[i]>=position){
                position=i;
                steps++;
                break;
            }             
        }
    }
    return steps;
}
}